core/controllers/admin.py

```diff
@@ -52,6 +80,31 @@ def test_super_admin(self, **kwargs):
     return test_super_admin
 
 
+def check_and_update_config_role(old_list, new_list, role):
+    """Changes roles for users in old_list and new_list. If user belongs to
+    new_list but not in old list, its role is changed to specified role. If a
+    user belongs to old list but not new list, its role is changed to
+    exploration editor.
+
+    Args:
+        old_list: list(str). Existing list of usernames with given
+            role.
+        new_list: list(str). Updated list of usernames with given role.
+        role: str. The role to be used for users in these lists.
+    """
+    for username in old_list:
```

Comment:
Btw, are you sure this works? E.g. if I set userA as a moderator and as a collection editor, will userA end up being a moderator after this change? If the latter change is made after the former then the end situation will be incorrect.

Perhaps a better approach is:
- Gather the union of the users in all the lists (both old and new, so that you can see the totality of who is affected)
- Compute the final role for each user (based on the "max" of the "new state" of the config property values)
- Update all the roles.